### PR TITLE
[WEBSITE-405] Use plugin-specific title and description in response metadata

### PR DESCRIPTION
**Pull request by timja**
Downstream of https://github.com/jenkins-infra/jenkins.io/pull/2674

Related to issue # WEBSITE-405

Summary of this pull request: 
SEO fixes to use specific title, and description

cc @halkeye  @zbynek @oleg-nenashev

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -35,6 +35,10 @@ app.use(jsPath, express.static('./dist/client'));
 app.engine('hbs', exphbs({extname: '.hbs'}));
 app.set('view engine', 'hbs');
 
+const defaultPluginTitle = 'Jenkins Plugins';
```
Comment (oleg-nenashev) [inline server.js]:
```suggestion
const defaultPluginSiteTitle = 'Jenkins Plugins';
```

File: server.js [hunk 2]
```diff
@@ -113,9 +119,17 @@ app.get('*', (req, res, next) => {
         const pluginSiteApiVersion = store.getState().data.info.commit.substring(0, 7);
         const reduxState = JSON.stringify(store.getState()).replace(/</g, '\\x3c');
         const pluginNotFound = req.url !== '/' && store.getState().ui.plugin === null;
+
+        const title = store.getState().ui.plugin && store.getState().ui.plugin.title ? `${store.getState().ui.plugin.title} - Jenkins plugin` : defaultPluginTitle;
+        const description = store.getState().ui.plugin && store.getState().ui.plugin.excerpt ? store.getState().ui.plugin.excerpt : defaultPluginDescription;
+        const opengraphImage = defaultPluginOpenGraphImage; // TODO add support for plugins to provide their own OG imag
```
Comment (oleg-nenashev) [inline server.js]: Add a ticket to https://issues.jenkins-ci.org/browse/WEBSITE-637 ?

File: server.js [hunk 3]
```diff
@@ -35,6 +35,10 @@ app.use(jsPath, express.static('./dist/client'));
 app.engine('hbs', exphbs({extname: '.hbs'}));
 app.set('view engine', 'hbs');
 
+const defaultPluginTitle = 'Jenkins Plugins';
+const defaultPluginDescription = 'Jenkins – an open source automation server which enables developers around the world to reliably build, test, and deploy their software';
```
Comment (oleg-nenashev) [inline server.js]: This description does not help in the case of plugins. I believe that almost all plugins already set a kind of description, but we might want to have a somethinf more explicit. Not a blocker

Comment (oleg-nenashev) [inline server.js]: I suggest changing the variable to indicate the name and description are not specific to the plugin site

```suggestion
const defaultPluginSiteDescription = 'Jenkins – an open source automation server which enables developers around the world to reliably build, test, and deploy their software';
```

Comment (timja) [reply server.js]: The description comes from the index.jelly so it's required AFAIK, this is used on pages that aren't plugins, i.e. at least the home page.

Happy to reword it, but this is what it was before

File: server.js (hunk 2 above)
Comment (timja) [reply server.js]: There already was one, added reference

**Review by oleg-nenashev — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
I will try to land it today

**Comment by zbynek** [comment]
@timja @oleg-nenashev does title change on navigation for you?

**Comment by timja** [comment]
> @timja @oleg-nenashev does title change on navigation for you?

no it doesn't struggling to fix it though,
figured out why I was getting so stumped earlier though, any errors in server.js were getting swallowed

**Comment by timja** [comment]
> @timja @oleg-nenashev does title change on navigation for you?

Fixed

**Comment by zbynek** [comment]
now the title is updated on navigation to a plugin, but not when you navigate back with the "fid plugins" link

**Comment by timja** [comment]
> now the title is updated on navigation to a plugin, but not when you navigate back with the "fid plugins" link

Seems to work for me?
Could you please upload a gif or be more specific on URLs / steps?

i.e. I did:
```
export HEADER_FILE=http://0.0.0.0:4242/template/index.html
yarn server

open http://localhost:5000

```

* type slack, enter
* Click Slack Notification
* See title updated
* Click Find plugins
* See title updated
* Click cucumber-slack-notifier
* See title updated

**Comment by zbynek** [comment]
* run
```
HEADER_FILE=https://gist.githubusercontent.com/zbynek/816b7cb8a613a4e1fe4a3ba5c160dab8/raw yarn server
```
* open http://localhost:5000/git
* click Find plugins -> title is still Git, expected Jenkins plugins
* click any plugin on index page -> title is OK
* click Find plugins again -> title reverts to git

If I start by visiting the index page, it works

**Comment by timja** [comment]
> * run
> 
> ```
> HEADER_FILE=https://gist.githubusercontent.com/zbynek/816b7cb8a613a4e1fe4a3ba5c160dab8/raw yarn server
> ```
> 
> * open http://localhost:5000/git
> * click Find plugins -> title is still Git, expected Jenkins plugins
> * click any plugin on index page -> title is OK
> * click Find plugins again -> title reverts to git
> 
> If I start by visiting the index page, it works

Fixed thanks for the testing

**Comment by zbynek** [comment]
@timja looks good now :tada:

**Comment by oleg-nenashev** [comment]
:shipit:

**Pull request merged**